irobotclient/request_handler.py

```diff
@@ -0,0 +1,161 @@
+"""
+Copyright (c) 2017 Genome Research Ltd.
+
+This program is free software: you can redistribute it and/or modify it
+under the terms of the GNU General Public License as published by the
+Free Software Foundation, either version 3 of the License, or (at your
+option) any later version.
+
+This program is distributed in the hope that it will be useful, but
+WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with this program. If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import requests
+import time
+import errno
+
+from datetime import datetime
+from collections import namedtuple
+
+from irobotclient.custom_exceptions import IrobotClientException
+
+"""
+Limit on how many times the same request is sent, including any alterations such as changes to the header.
+In theory a single request will take two attempts; one for the HEAD request to make sure the data exists and
+a second to GET the data.
+
+However, a request may generate a response to wait for the data to be fetched and try again later or to attempt
+a different authorisation method.
+
+Either way, the limit defined below helps prevent stack overflow and unnecessary requests when it is clear a request
+is not going to work.
+"""
+REQUEST_LIMIT = 10
+
+"""
+If a 202 response returns with not iRobot-ETA header then a default delay (in seconds) will be set.
+"""
+DEFAULT_REQUEST_DELAY = 600
```

Comment:
Implemented a function to get a default wait time from the environment or, failing that, hard coded at 600 seconds.